src/dune_rules/bootstrap_info.ml

```diff
@@ -64,11 +64,8 @@ let rule sctx ~requires_link =
         ])
     |> Action_builder.of_memo
   in
-  let externals =
```

Comment:
I understand that Dune generates the `thread.posix` every time, but I don't understand where it comes from. What is a change in the compiler?

Reply:
I don't understand why this is reported like this. Maybe @nojb knows the reason?

Reply:
I believe we are using findlib (or dune's version) for finding these external libraries. At some point ~ OCaml 5.2 we started getting `threads.posix` rather than both `unix` and `threads`.

Reply:
It is a bug in 153fcd1bc5c9f850a1f6b50e213a48d548c377ac, which inverted the condition. `threads.posix` is supposed to be filtered _out_.

Reply:
@nojb Good catch! I will open another PR with a fix.